CMake Error

I'm trying to run the hello_world sample, but I get these errors when I type `cmake -GNinja -DBOARD=qemu_x86 ..`:
```

CMake Deprecation Warning at /home/bruno_marques/zephyr/cmake/app/boilerplate.cmake:38 (cmake_pol
icy):                                                                                            
  The OLD behavior for policy CMP0000 will be removed from a future version                      
  of CMake.                                                                                      
                                                                                                 
  The cmake-policies(7) manual explains that the OLD behaviors of all                            
  policies are deprecated and that a policy should be set to OLD only under                      
  specific short-term circumstances.  Projects should be ported to the NEW                       
  behavior and not rely on setting a policy to OLD.                                              
Call Stack (most recent call first):                                                             
  CMakeLists.txt:2 (include)                                                                     
                                                                                                 
                                                                                                 
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.5.2", minimum required is "3.4
")                                                                                               
-- Selected BOARD qemu_x86                                                                       
Zephyr version: 1.12.99                                                                          
Parsing Kconfig tree in /home/bruno_marques/zephyr/Kconfig                                       
Using /home/bruno_marques/zephyr/boards/x86/qemu_x86/qemu_x86_defconfig as base                  
Merging /home/bruno_marques/zephyr/samples/hello_world/prj.conf                                  
/opt/zephyr-sdk/sysroots/x86_64-pokysdk-linux/usr/bin/i586-zephyr-elf/i586-zephyr-elf-gcc: 1: /op
t/zephyr-sdk/sysroots/x86_64-pokysdk-linux/usr/bin/i586-zephyr-elf/i586-zephyr-elf-gcc: Syntax er
ror: ")" unexpected                                                                              
CMake Error at /home/bruno_marques/zephyr/cmake/compiler/gcc.cmake:48 (string):                  
  string sub-command REGEX, mode REPLACE needs at least 6 arguments total to                     
  command.                                                                                       
Call Stack (most recent call first):                                                             
  /home/bruno_marques/zephyr/cmake/toolchain.cmake:40 (include)                                  
  /home/bruno_marques/zephyr/cmake/app/boilerplate.cmake:250 (include)                           
  CMakeLists.txt:2 (include)                                                                     
                                                                                                 
                                                                                                 
/opt/zephyr-sdk/sysroots/x86_64-pokysdk-linux/usr/bin/i586-zephyr-elf/i586-zephyr-elf-gcc: 1: /op
t/zephyr-sdk/sysroots/x86_64-pokysdk-linux/usr/bin/i586-zephyr-elf/i586-zephyr-elf-gcc: Syntax er
ror: ")" unexpected                                                                              
CMake Error at /home/bruno_marques/zephyr/cmake/compiler/gcc.cmake:48 (string):                  
  string sub-command REGEX, mode REPLACE needs at least 6 arguments total to                     
  command.                                                                                       
Call Stack (most recent call first):                                                             
  /home/bruno_marques/zephyr/cmake/toolchain.cmake:40 (include)                                  
  /home/bruno_marques/zephyr/cmake/app/boilerplate.cmake:250 (include)                           
  CMakeLists.txt:2 (include)                                                                     
                                                                                                 
                                                                                                 
/opt/zephyr-sdk/sysroots/x86_64-pokysdk-linux/usr/bin/i586-zephyr-elf/i586-zephyr-elf-gcc: 1: /op
t/zephyr-sdk/sysroots/x86_64-pokysdk-linux/usr/bin/i586-zephyr-elf/i586-zephyr-elf-gcc: Syntax er
ror: ")" unexpected                                                                              
CMake Error at /home/bruno_marques/zephyr/cmake/extensions.cmake:1081 (message):                 
  No such file or directory: LIBGCC_FILE_NAME: ''                                                
Call Stack (most recent call first):                                                             
  /home/bruno_marques/zephyr/cmake/compiler/gcc.cmake:100 (assert_exists)                        
  /home/bruno_marques/zephyr/cmake/toolchain.cmake:40 (include)                                  
  /home/bruno_marques/zephyr/cmake/app/boilerplate.cmake:250 (include)                           
  CMakeLists.txt:2 (include)                                                                     
                                                                                                 
-- Configuring incomplete, errors occurred!           
```

--- a/cmake/compiler/gcc.cmake
+++ b/cmake/compiler/gcc.cmake
@@ -24,6 +24,19 @@ TOOLCHAIN_HOME: ${TOOLCHAIN_HOME}
 ")
 endif()
 
+execute_process(
+  COMMAND ${CMAKE_C_COMPILER} --version
+  RESULT_VARIABLE ret
+  OUTPUT_QUIET
+  ERROR_QUIET
+  )
+if(ret)
+  message(FATAL_ERROR "Executing the below command failed. Are permissions set correctly?
+'${CMAKE_C_COMPILER} --version'
+"
+    )
+endif()
+
 if(CONFIG_CPLUSPLUS)
   set(cplusplus_compiler ${CROSS_COMPILE}${C++})
 else()
@@ -45,7 +58,7 @@ foreach(file_name include include-fixed)
     COMMAND ${CMAKE_C_COMPILER} --print-file-name=${file_name}
     OUTPUT_VARIABLE _OUTPUT
     )
-  string(REGEX REPLACE "\n" "" _OUTPUT ${_OUTPUT})
+  string(REGEX REPLACE "\n" "" _OUTPUT "${_OUTPUT}")
 
   if(MSYS)
     # TODO: Remove this when

